internal/pkg/types/message.go

```diff
@@ -10,16 +10,22 @@ import (
 	"github.com/filecoin-project/go-amt-ipld"
 	"github.com/ipfs/go-cid"
 	"github.com/ipfs/go-datastore"
-	"github.com/ipfs/go-ipfs-blockstore"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
 	cbor "github.com/ipfs/go-ipld-cbor"
 	ipld "github.com/ipfs/go-ipld-format"
 	errPkg "github.com/pkg/errors"
 
 	"github.com/filecoin-project/go-filecoin/internal/pkg/encoding"
 	"github.com/filecoin-project/go-filecoin/internal/pkg/vm/address"
-	"github.com/whyrusleeping/cbor-gen"
+	typegen "github.com/whyrusleeping/cbor-gen"
 )
 
+// MethodID is an identifier of a method (in an actor).
+type MethodID UVarint
```

Comment:
I like that we have an explicit type name for method ids, distinguishing them from other integers. Help me understand the encoding. A MethodID in the message is supposed to be encoded as a variable-length integer. This new `UVarint` is represented as a Go `uint` and is not encoded with variable-length encoding as far as I can see.

If you want to defer the varint encoding for follow-up, that's ok. Just use 'uint' here and leave a comment referencing a GitHub issue for fixing the encoding. Indirecting through this new `UVarint` isn't adding anything, and just obscures the fact that it's not in fact varint encoded. Since this is the only reference to `UVarint` outside its definition, inserting this indirection again later will be trivial (and, I would suggest, be named after its runtime representation if we chose a fixed-size).

If you want to use the correct encoding now, use the existing Uint64 type which implements that. Yes, 64-bits is excessive for runtime representation, so we could also file an issue to replace this alias with a Uint32 or even Uint16 later, purely as an optimization.

Reply:
I disagree on principle, we can discuss it offline. 
I removed it so we can get this merged and move on.